skypy/power_spectrum/_camb.py

```diff
@@ -72,29 +77,31 @@ def camb(wavenumber, redshift, cosmology, A_s, n_s):
                        omch2=cosmology.Odm0 * h2,
                        omk=cosmology.Ok0,
                        TCMB=cosmology.Tcmb0.value,
-                       mnu=np.sum(cosmology.m_nu.value),
+                       mnu=0.,  # for now, need to solve agreement with CLASS
                        standard_neutrino_neff=cosmology.Neff
                        )
 
-    # camb requires redshifts to be in decreasing order
-    redshift_order = np.argsort(redshift)[::-1]
+    # camb interpolator requires redshifts to be in increasing order
+    redshift_order = np.argsort(redshift)
 
     pars.InitPower.ns = n_s
     pars.InitPower.As = A_s
 
+    pars.Reion.redshift = z_reio
+
     pars.set_matter_power(redshifts=list(redshift[redshift_order]),
                           kmax=np.max(wavenumber))
 
     pars.NonLinear = model.NonLinear_none
 
-    results = get_results(pars)
-
     k = wavenumber * (1. / u.Mpc)
 
-    k_h = k.to((u.littleh / u.Mpc), u.with_H0(cosmology.H0))
+    pk_interp = get_matter_power_interpolator(pars,
+                                              nonlinear=False,
+                                              hubble_units=False, k_hunit=False,
+                                              kmax=np.max(k.value),
+                                              zmax=redshift.max())
 
-    kh, z, pzk = results.get_matter_power_spectrum(minkh=np.min(k_h.value),
-                                                   maxkh=np.max(k_h.value),
-                                                   npoints=len(k_h.value))
+    pzk = pk_interp.P(redshift[redshift_order], k.value)
```

Comment:
Use arrays as provided.
```suggestion
    pzk = pk_interp.P(redshift, wavenumber)
```